code/script.js

```diff
@@ -222,104 +274,158 @@ const generateBoard = () => {
+      )
+    } else {
+      alert(
+        `No, the person doesn't wear ${value}! Remove all people that wear ${value}`
+      )
+    }
+  } if (category === 'shoes') {
+    if (keep) {
+      alert(
+        `Yes, the person wear ${value} shoes! Keep all people that wears ${value} shoes`
+      )
+    } else {
+      alert(
+        `No, the person doesn't wear ${value} shoes! Remove all people that wears ${value} shoes`
+      )
     }
   }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+  if (category === 'hair' || category === 'eyes') {
+    if (keep) {  
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+    } else {
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+    }
 
-    for accessories and other
+  // else statement for accessories, sweater and face we need to check if the value is included in an array.
+  } else {
+    if (keep) {
       charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
+    } else {
       charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
+    }
+  }
+  generateBoard()
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
+start()
+
+
+// This function happens when user want to make a guess
+// If the player wants to guess, invoke the checkMyGuess function
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  const madeGuess = confirm(`Are you sure you want to guess on ${personToConfirm}?`) 
+ if (madeGuess) {
+  checkMyGuess(personToConfirm)   
+}
 }
 
 // If you confirm, this function is invoked
+
 const checkMyGuess = (personToCheck) => {
-  // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+  if (personToCheck === secret.name) {                                              // 1. Check if the personToCheck is the same as the secret person's name
+    winOrLoseText.innerHTML = `You are the winner! ${secret.name} is correct!`      // 2. Set a Message to show in the win or lose section accordingly
+    winOrLose.style.display = 'block',                                              // 3. Show the win or lose section
+    winOrLoseAudio.innerHTML = `<audio src="./02" autoplay></audio>`     // Playing winning sound. 
```

Comment:
The winOrLoseAudio variable is undefined. Perhaps you need to add an HTML element to this variable at the top?
One of the files also seems to be a problem. The link to fanfare.mp3 works but not ./02 So maybe the file is a problem here too?